components/chart-file-set.js

```diff
@@ -0,0 +1,328 @@
+      if (selectedMonthButton) {
+        selectedMonthButton.focus();
+      }
+    }
+  }, [isSelectorOpen]);
+
+  return (
+    <div id="month-selector" className={className}>
+      <button
+        id="month-selector-reference"
+        aria-label={`Filter the chart for ${
+          selectedMonthFormatted === "All"
+            ? "all months"
+            : selectedMonthFormatted
+        }`}
+        aria-expanded={isSelectorOpen}
+        aria-haspopup="listbox"
+        className="h-6 rounded border border-button-primary bg-button-primary px-2 text-xs text-button-primary"
+        ref={refs.setReference}
+        {...getReferenceProps()}
+      >
+        {selectedMonthFormatted}
+      </button>
+      {isSelectorOpen && (
+        <div
+          id="month-selector-list"
+          data-testid="month-selector-list"
+          aria-labelledby="month-selector-reference"
+          className="w-64"
+          style={floatingStyles}
+          ref={refs.setFloating}
+          {...getFloatingProps()}
+        >
+          <ListSelect
+            value={selectedMonth}
+            onChange={monthSelection}
+            className="shadow-lg [&>div]:max-h-52"
+          >
+            {months.map((month) => {
+              const displayedMonth = formatMonth(month);
+              return (
+                <ListSelect.Option
+                  key={month}
+                  id={month}
+                  label={displayedMonth}
+                >
+                  <div className="text-left">{displayedMonth}</div>
+                </ListSelect.Option>
+              );
+            })}
+          </ListSelect>
+        </div>
+      )}
+    </div>
+  );
+}
+
+MonthSelector.propTypes = {
+  // FileSet objects to collect months for
+  fileSets: PropTypes.arrayOf(PropTypes.object).isRequired,
+  // Currently selected month
+  selectedMonth: PropTypes.string.isRequired,
+  // Called when the user selects a month
+  setSelectedMonth: PropTypes.func.isRequired,
+  // Tailwind CSS classes to apply to the selector
+  className: PropTypes.string,
+};
+
+/**
+ * Display a chart of MeasurementSets by lab and summary.
+ */
+export default function ChartFileSet({ fileSets }) {
+  // Currently selected month to filter the chart by
+  const [selectedMonth, setSelectedMonth] = useState("All");
+
+  // Get the dark mode setting from the global context for cases we can't use Tailwind CSS.
+  const { darkMode } = useContext(GlobalContext);
+  const legendTextColor = darkMode.enabled ? "#ffffff" : "#000000";
+
+  // Filter the file sets by the selected month, then convert to a form the Nivo bar chart can use.
+  const selectedFileSets = filterFileSetsByMonth(fileSets, selectedMonth);
+  const { fileSetData, maxCount } = convertDataToNivo(selectedFileSets);
+
+  // The legend should only appear while the user is logged in.
+  const { isAuthenticated } = useAuth0();
+  const legendProps = isAuthenticated
```

Comment:
Why legend is hidden if not logged in? I don't feel that this is sensitive data.

Reply:
The legend shows the color differences between released and non-released file sets in the chart. But while logged out you only see released file sets, so Meenakshi felt it didn’t make sense to show this legend when you’d only ever see one color in the chart.